### PR TITLE
Remove job from logs

### PR DESCRIPTION
**Pull request by runcom**
Signed-off-by: Antonio Murdaca me@runcom.ninja

### CODE REVIEW

**Comment by duglin** [comment]
LGTM

File: daemon/logs.go [hunk 1]
```diff
@@ -10,35 +10,30 @@ import (
...
-		stdout = job.GetenvBool("stdout")
-		stderr = job.GetenvBool("stderr")
-		tail   = job.Getenv("tail")
-		follow = job.GetenvBool("follow")
-		times  = job.GetenvBool("timestamps")
 		lines  = -1
 		format string
 	)
-	if !(stdout || stderr) {
+	if !(config.Stdout != nil || config.Stderr != nil) {
```
Comment (LK4D4) [inline daemon/logs.go]: This is simply `config.Stdout == nil && config.Stderr == nil`

Comment (runcom) [inline daemon/logs.go]: sure

File: api/server/server.go [hunk 2]
```diff
@@ -557,43 +557,47 @@ func getContainersLogs(eng *engine.Engine, version version.Version, w http.Respo
...
+	stdout, stderr := toBool(r.Form.Get("stdout")), toBool(r.Form.Get("stderr"))
 	if !(stdout || stderr) {
 		return fmt.Errorf("Bad parameters: you must choose at least one stream")
 	}
-	if err = inspectJob.Run(); err != nil {
+
+	name := vars["name"]
+	d := getDaemon(eng)
+
+	cont, err := d.Get(name)
```
Comment (LK4D4) [inline api/server/server.go]: You getting container here and inside daemon. Seems like you have no choice without significant refactoring, but it is not consistent with your previous PR.

Comment (LK4D4) [inline api/server/server.go]: Actualy I think you can do this with additional config parameter, like `muxStreams`. Api cares only about `w`.

Comment (runcom) [inline api/server/server.go]: @LK4D4 how do you suggest to go with this? moving container check inside d.ContainerLogs?

Comment (LK4D4) [inline api/server/server.go]: There is already check inside `ContainerLogs`. I was mostly concerned about `c.Config.Tty`.

Comment (runcom) [inline api/server/server.go]: `container.Config.Tty` value is passed also in cli code

Comment (runcom) [inline api/server/server.go]: https://github.com/docker/docker/blob/master/api/client/logs.go#L54

Comment (runcom) [inline api/server/server.go]: what if I do the multiplexing inside d.ContainerLogs? would it be ok? I'll just pass a writeFlusher to ContainerLogs then @LK4D4

Comment (LK4D4) [inline api/server/server.go]: If it'll allow you to remove double check of `d.Get`, then okay.

**Comment by runcom** [comment]
@LK4D4 take a look, I've moved the check and mux inside `d.ContainerLogs` (I think something can be further simplified though)

File: daemon/logs.go [hunk 3]
```diff
@@ -10,40 +10,52 @@ import (
...
+		config.Tail = "all"
 	}
+
 	container, err := daemon.Get(name)
 	if err != nil {
 		return err
 	}
+
+	var (
+		outStream = config.OutStream
```
Comment (runcom) [inline daemon/logs.go]: this is not really needed but I think make code more readable from here...

File: daemon/logs.go [hunk 4]
```diff
@@ -10,40 +10,52 @@ import (
...
+		errStream io.Writer
+	)
+	if !container.Config.Tty && config.ApiVersion.GreaterThanOrEqualTo("1.6") {
```
Comment (LK4D4) [inline daemon/logs.go]: I'm pretty sure that version check is not necessary. I blindly copied it from attach, but `/logs/` endpoint isn't even exists before "1.6"

File: integration-cli/docker_api_containers_test.go [hunk 5]
```diff
@@ -537,7 +538,7 @@ func TestPostContainerBindNormalVolume(t *testing.T) {
 	}
 
 	bindSpec := map[string][]string{"Binds": {fooDir + ":/foo"}}
-	_, err = sockRequest("POST", "/containers/two/start", bindSpec)
+	_, _, err = sockRequest("POST", "/containers/two/start", bindSpec)
```
Comment (LK4D4) [inline integration-cli/docker_api_containers_test.go]: There are a lot of such places. Maybe makes sense to create issue about checking status there?

File: api/server/server.go [hunk 6]
```diff
@@ -557,43 +557,30 @@ func getContainersLogs(eng *engine.Engine, version version.Version, w http.Respo
...
 	if !(stdout || stderr) {
 		return fmt.Errorf("Bad parameters: you must choose at least one stream")
 	}
-	if err = inspectJob.Run(); err != nil {
-		return err
-	}
 
-	var outStream, errStream io.Writer
+	var outStream io.Writer
 	outStream = utils.NewWriteFlusher(w)
```
Comment (LK4D4) [inline api/server/server.go]: let's just pass `utils.NewWriteFlusher(w)` directly as `OutStream` to config.

File: integration-cli/docker_api_containers_test.go (hunk 5 above)
Comment (runcom) [inline integration-cli/docker_api_containers_test.go]: Cool

File: api/server/server.go (hunk 6 above)
Comment (runcom) [inline api/server/server.go]: And just use `w` directly below?

**Comment by LK4D4** [comment]
LGTM

File: integration-cli/docker_api_containers_test.go (hunk 5 above)
Comment (runcom) [inline integration-cli/docker_api_containers_test.go]: I'll create the issue for this one @LK4D4

**Pull request merged**